beacon.go

```
@@ -17,110 +17,174 @@
+		writeError(w, err)
+		return
 	}
 
-	coord, err := strconv.ParseInt(r.FormValue("coordinate"), 10, 64)
+	refName, allele, coord, err := parseInput(r)
 	if err != nil {
-		return nil, "", err
+		writeError(w, err)
+		return
 	}
 
-	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
-	search.Start = coord
-	search.End = search.Start + 1
+	ctx := appengine.NewContext(r)
+	exists, err := genomeExists(ctx, refName, allele, coord)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
 
-	if search.ReferenceName == "" {
-		return nil, "", errors.New("missing reference name")
+	if err := writeResponse(w, exists); err != nil {
+		writeError(w, err)
+		return
 	}
-	if allele == "" {
-		return nil, "", errors.New("missing allele")
+}
+
+func genomeExists(ctx context.Context, refName string, allele string, coord int64) (bool, error) {
+	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
+	query := fmt.Sprintf(`
+		SELECT count(v.reference_name) as count
+		FROM %s as v
+		WHERE reference_name='%s'
+			AND v.start <= %d AND %d < v.end
+	 	 	AND reference_bases='%s'
+		LIMIT 1`,
+		fmt.Sprintf("`%s`", config.tableId),
+		refName,
+		coord,
+		coord+1,
+		allele)
+
+	bqclient, err := bigquery.NewClient(ctx, config.projectID)
+	if err != nil {
+		return false, newDataAccessError("Creating a BigQuery client", err)
+	}
+	it, err := bqclient.Query(query).Read(ctx)
+	if err != nil {
+		return false, newDataAccessError("Querying database", err)
 	}
 
-	return search, allele, nil
+	type Result struct {
+		Count int
+	}
+	var result Result
+	if err := it.Next(&result); err != nil {
+		return false, newDataAccessError("Reading query result", err)
+	}
+	return result.Count > 0, nil
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	c := appengine.NewContext(r)
-	search, allele, err := requestToSearch(r)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+func validateServerConfig() error {
```

You should just return a bare error from this function

And again, you don't need to use any special error functions here: a server error will be the default kind of error.  You only need to use a specific error type when you want a non-500 error code.

---

done